src/printer.js

```diff
@@ -1559,7 +1559,7 @@ function genericPrintNoParens(path, options, print) {
         );
       }
 
-      return concat(parts);
+      return group(concat(parts));
```

Comment:
By moving the group one level outside, it now first breaks on the arguments and then on the generics. Without this change, it broke a lot of tests with results like this one ("deleted" is the behavior we now have):

```diff
-declare function my_filter<T, P: $Pred<1>>(
-  v: Array<T>,
-  cb: P
-): Array<$Refine<T, P, 1>>;
+declare function my_filter<T, P: $Pred<1>>(v: Array<T>, cb: P): Array<
+  $Refine<T, P, 1>
+>;
```